pycode/memilio-epidata/memilio/epidata_test/test_epidata_get_Case_Data.py

```diff
@@ -0,0 +1,1025 @@
+        no_raw = False
+        impute_dates = False
+        make_plot = False
+        moving_average = 0
+        split_berlin = False
+        rep_date = False
+
+        directory = os.path.join(out_folder, 'Germany/')
+        gd.check_dir(directory)
+
+        self.write_case_data_arcgis(directory)
+        self.write_case_data_not_all_states(directory)
+        # check if expected files are written
+        self.assertEqual(len(os.listdir(directory)), 2)
+
+        # test case where all files are incomplete
+        mock_loadCsv.return_value = pd.read_json(
+            os.path.join(directory, "CaseDataNotFull.json"))
+        mock_loadGeojson.return_value = pd.read_json(
+            os.path.join(directory, "CaseDataNotFull.json"))
+        with self.assertRaises(FileNotFoundError) as error:
+            gcd.get_case_data(
+                read_data, file_format, out_folder, no_raw, impute_dates,
+                make_plot, moving_average, split_berlin, rep_date)
+        self.assertEqual(
+            str(error.exception),
+            "Something went wrong, dataframe is empty for csv and geojson!")
+
+        mock_loadGeojson.assert_called_once()
+        mock_loadCsv.assert_called()
+
+        # test case where csv files are incorrect
+        mock_loadCsv.side_effect = [pd.DataFrame(), pd.read_json(
+            os.path.join(directory, "CaseDataNotFull.json"))]
+        mock_loadGeojson.return_value = pd.read_json(
+            os.path.join(directory, "CaseDataArcgis.json"))
+
+        gcd.get_case_data(
+            read_data, file_format, out_folder, no_raw, impute_dates,
+            make_plot, moving_average, split_berlin, rep_date)
+
+        mock_loadGeojson.assert_called()
+        mock_loadCsv.assert_called()
+
+        # check if expected files are written
+        # (14 written files + 2 source files: "CaseDataNotFull.json" and "CaseDataArcgis.json")
+        self.assertEqual(len(os.listdir(directory)), 14+2)
+
+        # test output files
+        file = "cases_all_germany.json"
+        f_read = os.path.join(directory, file)
+        df = pd.read_json(f_read)
+
+        file = 'cases_infected.json'
+        f_read = os.path.join(directory, file)
+        df_infected = pd.read_json(f_read)
+
+        file = 'cases_deaths.json'
+        f_read = os.path.join(directory, file)
+        df_deaths = pd.read_json(f_read)
+
+        data_list = df.columns.values.tolist()
+        self.assertEqual(
+            data_list, ["Date", "Confirmed", "Deaths", "Recovered"])
+        self.assertEqual(df[(df['Date'] == "2020-06-15")]['Confirmed'].item(),
+                         df_infected[(df_infected['Date'] == "2020-06-15")]['Confirmed'].item())
+        self.assertEqual(df[(df['Date'] == "2020-06-15")]
+                         ['Confirmed'].item(), 3)
+        self.assertEqual(
+            df[(df['Date'] == "2021-01-06")]['Deaths'].item(),
+            df_deaths[(df_deaths['Date'] == "2021-01-06")]['Deaths'].item())
```

Comment:
I introduced different test sets depending on the source (read/github/arcgis). In the test sets for data from github and arcgis I did not inculde the 19 columns defined above (line 51ff) for readability reasons. Since this data was excluded the dates for them are missing in the test set, so I needed to change the tested dates to dates from the data in test_epidata_getCaseData_data_github.json/..._arcgis.json and adapt the values accordingly.